### DISCUSSION TITLE
Update mechanism

### DISCUSSION THREAD

**Discussion by ppeterrr** [qdm12/gluetun#1108] [Q&A]
hey,

I've been trying to figure out how to automatically switch the endpoint ip I'm using with a custom wireguard config for some time now.

Now I'm reading an option here, which I didn't see before. The Update mechanism, reading from the servers.json file.

It's not entirely clear to me, so I thought I'd open a topic here.

Am I understanding correctly, if I edit the json with my custom wireguard settings, and create a list below that with different endpoint ips.
Then gluetun will grab an IP further in the list after each update interval?

Thanks

**Comment by qdm12** [comment]
The update mechanism and servers.json is really only for supported vpn providers, it won't work for the customer provider.

> how to automatically switch the endpoint ip

Why do you want to change the endpoint ip?

**Reply by ppeterrr** [reply]
I use air vpn, as a custom provider. They provide about 50 or more servers around the globe. All with their own endpoint. In the config of gluetun I only change the endpoint ip. Therefore all connected clients obtain an ip from that region.

Normally I change that manually, once in a while. But I'd like to have that done automatically. Something like a new region every week.

Perhaps I'm not quite right, and things are not clear for me. But I think it is wiser to switch continuously with the VPN connection.

I was already thinking about creating multiple containers with different settings in my docker environment on the synology nas. And then have it stop and start in a cycle with a script.

Maybe i can learn something :) I am totally not sure if i am doing the right thing.

**Reply by Rohaq** [reply]
Good news! Scanning through their site,  I found they have a public API endpoint that can return pretty much all the information someone might need to add AirVPN support! 😄 

https://airvpn.org/api/status/

They also have a handy testing tool for their API here: https://airvpn.org/apisettings/

**Reply by Rohaq** [reply]
![image](https://user-images.githubusercontent.com/246865/185462754-b1c83dc7-01eb-4046-a7b8-40c1b7b2182a.png)

**Reply by qdm12** [reply]
I'll look into it. If it's natively supported, you could just have a cron job curling the gluetun control server to restart the vpn and that would pick another server at random (according to filters set, if any). You wouldn't have to restart connected containers either.

**Reply by ppeterrr** [reply]
> I'll look into it. If it's natively supported, you could just have a cron job curling the gluetun control server to restart the vpn and that would pick another server at random (according to filters set, if any). You wouldn't have to restart connected containers either.

That would be great, in the meanwhile i'll try to figure out the stuff about the control server.
Would it be that "PUT" status aint working because it is not specified in the api, since i am running a custom profile for now?

`C:\Users\peter>curl http://192.168.1.42:8001/v1/dns/status
{"status":"running"}

C:\Users\peter>curl -X PUT -H "Content-Type: application/json" -d '{"status":"stopped"}' http://192.168.1.42:8001
unversioned API: only supports GET method`

**Reply by qdm12** [reply]
What are you trying to PUT on? The API is documented at https://github.com/qdm12/gluetun/wiki/HTTP-control-server

You can restart gluetun with the API, but that will just re-use your previous openvpn config (unless you bind mount it, I'm not 100% sure?).

**Reply by ppeterrr** [reply]
> What are you trying to PUT on? The API is documented at https://github.com/qdm12/gluetun/wiki/HTTP-control-server
> 
> You can restart gluetun with the API, but that will just re-use your previous openvpn config (unless you bind mount it, I'm not 100% sure?).

Thanks, i am just testing and trying to learn something. I am aware it will not work with the custom config.

**Reply by ppeterrr** [reply]
> I'll look into it. If it's natively supported, you could just have a cron job curling the gluetun control server to restart the vpn and that would pick another server at random (according to filters set, if any). You wouldn't have to restart connected containers either.

Thanks a lot for adding airvpn, also this workaround you suggested is working like a charm.
Hopping around the world now :)

**Comment by ppeterrr** [comment]
@Rohaq Thanks for pointing that out.

That needs some research on my part. It does seem a bit complex.
